src/ripple/app/tx/impl/NFTokenAcceptOffer.cpp

```diff
@@ -230,7 +271,22 @@ NFTokenAcceptOffer::pay(
     if (amount < beast::zero)
         return tecINTERNAL;
 
-    return accountSend(view(), from, to, amount, j_);
+    auto const result = accountSend(view(), from, to, amount, j_);
+
+    // After this amendment, if any payment would cause a non-IOU-issuer to
+    // have a negative balance, or an IOU-issuer to have a positive balance in
+    // their own currency, we know that something went wrong. This was
+    // originally found in the context of IOU transfer fees. Since there are
+    // several payouts in this tx, just confirm that the end state is OK.
```

Comment:
Note: negative balances are *not* a problem. Trustlines are bi-directional, and the issuer *might* have configured the trustline so that you can go into debt. The problem is a balance that _exceeds_ the limit.

Reply:
(@ledhed2222 wrote this comment [here](https://github.com/ledhed2222/rippled/commit/36c87451275ebd4dfdbb7972d5460555e7b49faf))

@ledhed2222 if it makes sense to you, could you clarify this comment? You can do so in a new PR with `feature/nft-fixes`.

(or, you can push a commit on `feature/nft-fixes`, if permissions allow. Either way is fine)